cdi/src/main/java/io/smallrye/config/inject/ConfigProducerUtil.java

```diff
@@ -107,6 +111,36 @@ private ConfigProducerUtil() {
         return resolveConverterForIndexed(type, config, indexedConverter).convert(" ");
     }
 
+    /**
+     * Convert the entries whose key starts with the given name into a {@code Map} using the converters
+     * corresponding to the type arguments extracted from the given type.
+     * 
+     * @param name the name of the map that will be used as prefix.
+     * @param type the type of the map to build.
+     * @param config the configuration from which the entries are extracted.
+     * @param <T> the type of the object built by the method.
+     * @return a {@code Map} containing the converted entries that could be found in case the given
+     *         type is a type of map, an empty map otherwise.
+     */
+    private static <T> T convertValuesToMap(String name, Type type, Config config) {
+        Map map = new HashMap();
+        final String prefix = name.endsWith(".") ? name : name + ".";
+        if (type instanceof ParameterizedType) {
+            ParameterizedType paramType = (ParameterizedType) type;
+            Type[] typeArgs = paramType.getActualTypeArguments();
+            for (String pName : config.getPropertyNames()) {
+                if (pName.startsWith(prefix)) {
```

Comment:
What if you have multiple subnamespaces? For instance:

```
my.map.key=key
my.map.value=value
my.map.sub.something=1234
```

Reply:
@radcortez You mean that in that case you would expect only the keys `key` and `value` in the map with `@ConfigProperty(name = "my.map")`?

Reply:
I guess this was not a good example :) 

My point was about keys after the map that have more that one level? Should the key be included as `x.y`? Should it be ignored because it is composed?

Reply:
I understand your point, for me since there many unspecified  aspects like this one, I wanted to keep it simple by making a "flat map". 
So in your example  If I have the following code:

```
@ConfigProperty(name = "my.map")
Map<String, String> map;
```

I expect to have key/key, value/value and sub.something/1234 in my map as pairs of String. 

Subnamespaces could make sense if we wanted to support complex mapping like:

```
my.map.c1.name=John
my.map.c1.age=33
```

The target class:
```
class Customer {
   String name;
   int age;
}
```

The mapping definition

```
@ConfigProperty(name = "my.map")
Map<String, Customer> map;
```

I could then expect to have one entry inside my map whose key is `c1` but I think that we would be too far from the current logic of the original spec where the value of all fields annotated with `@ConfigProperty` should be the value of the property converted into the target type.

Don't you agree?

Reply:
I don't think we should support complex mapping (well we could, since this is already supported via @ConfigMapping, but I believe it would require a lot of extra work)

What I'm trying to define here is what should be the behaviour in a case we find a subnamespace when injecting a map. I would say that we should ignore these properties.

Reply:
@radcortez Fixed